conbench/entities/commit.py

```diff
@@ -197,34 +191,17 @@ def __init__(self):
     def get_default_branch(self, name):
         url = f"{GITHUB}/repos/{name}"
         response = self._get_response(url)
-        if response:
-            return response["default_branch"]
-        return "main"
-
-    def get_commits(self, name, sha):
-        if sha in self.test_commits:
-            return self.test_commits
+        if not response:
+            return None
 
-        commits = []
+        if response["fork"]:
+            org = response["source"]["owner"]["login"]
+            branch = response["source"]["default_branch"]
+        else:
+            org = response["owner"]["login"]
+            branch = response["default_branch"]
 
-        # Grabs the last 1000 commits to the default branch. TODO: If the pull
-        # request is old, the parent may not be in the last 1000 commits.
-        branch = self.get_default_branch(name)
-        url = f"{GITHUB}/repos/{name}/commits?sha={branch}&per_page=100"
-        response = self._get_response(url)
-        if response:
-            commits = self._parse_commits(response)
-            if sha in commits:
-                return commits
-            for page in range(2, 11):
```

Comment:
Removing this makes conbench go so much faster, haha

Reply:
But surely pinging a giant API over and over isn't _that_ slow 😂 😭

Reply:
I ended up removing `@pytest.mark.slow` from these tests because now they're faster than some other tests, haha